Revision: 34610

ConnellyJHW opened the pull request:
【Task】om的业务关联改为支持多个相同业务关联
om的业务关联改为支持多个相同业务关联，业务之间是否支持多个关联，交由业务本身校验 

修改变更： 
1. om svc 关联表增加关联名Name字段，取消From和To的唯一索引，增加Name的唯一索引 
2. om svc Rest接口 create relationship增加参数Name 
```
{ 
   cmd: "create relationship", 
   Name: "xxxx", 
   From: "xxx", 
   To: "xxx", 
   Options: { 
      xxx: xxx, 
      ...... 
   } 
} 
```

3. om svc Rest接口 remove relationship移除参数From和To，增加参数Name 
```
{ 
   cmd: "remove relationship", 
   Name: "xxxx" 
} 
```

4. om agent关联业务和解除关联操作根据 om svc的参数做相关修改 
5. sac页面根据om svc的Rest接口变更做相关修改

FlorentinoJH commented:
请处理

FlorentinoJH commented:
把相应接口变更的，都列上.

BjornWGG commented:
列了， 就是 om svc Rest接口 create relationship和remove relationship

BjornWGG commented:
OM svc和OM agent的修改： 
Message: 
3316 
om的业务关联改为支持多个相同业务关联 
```
Modified : /trunk/SequoiaDB/engine/include/omDef.hpp 
Modified : /trunk/conf/script/createRelationship/sequoiasql-oltp.js 
Modified : /trunk/conf/script/removeRelationship/sequoiasql-oltp.js 
Modified : /trunk/SequoiaDB/engine/omsvc/omCommand.hpp 
Modified : /trunk/SequoiaDB/engine/omsvc/omCommandTool.cpp 
Modified : /trunk/SequoiaDB/engine/omsvc/omCmdRelationship.cpp 
Modified : /trunk/SequoiaDB/engine/omsvc/omCommandTool.hpp 
```

BjornWGG commented:
请修改sac网页

ConnellyJHW commented:
sac前端部分 

【提交记录】 
Message: 
3316 sac 
1.关联业务新增支持多个相同业务关联，增加参数Name，DbName。 
2.调整部署页面关联业务信息的显示格式。 
```
Modified : /trunk/SequoiaDB/web/app/controller/Deploy/Index.js 
Modified : /trunk/SequoiaDB/web/app/language/English.json 
Modified : /trunk/SequoiaDB/web/app/template/Deploy/Index.html 
```

【CheckList】 
检查项---------------------------------------------------------结论（NA/N/Y） 
修改代码是否全部提交SVN主干和分支版本 
且符合代码规范--------------------------------------------Y 主干 
是否用例全部通过-----------------------------------------N 手工测试 
修改代码是否通过检视------------------------------------N 
资料是否变更-----------------------------------------------N 
接口是否变更-----------------------------------------------N 
相关驱动是否变更或添加---------------------------------NA 
是否保持向前兼容------------------------------------------Y 
是否影响安装部署和监控---------------------------------N 
是否影响性能-----------------------------------------------N 
sdb shell帮助信息是否更新------------------------------NA

FlorentinoJH commented:
请安排测试

FlorentinoJH commented:
关联业务新增支持多个相同业务关联，增加参`数Name，DbName。 ===> DBName`

FlorentinoJH commented:
暂时保持 DbName。

EmilianLLJH commented:
请验证

BrigidLLX commented:
验证已完成，请关闭 
【验证场景】 
主要测试sdb业务和ssql业务数据库之间的创建删除关联，同时验证OM关联表上的Name字段唯一索引以及rest接口create relationship和remove relationship的变更 
1、相同sdb业务关联同一ssql业务下的不同数据库 
2、相同sdb业务关联不同ssql业务下的同名数据库 
3、不同sdb业务关联相同ssql业务下的同一数据库 
4、创建删除关联，检查OM关联表上的记录，检查关联表的索引 
5、创建删除关联，检查rest接口的消息结构 
6、指定不存在的数据库，创建关联 
7、创建已存在的关联 
【测试用例】 
TestLink： 
sequoiaDB/OM测试/OM支持SequoiaSQL-OLTP/关联SequoiaSQL-OLTP业务与SequoiaDB业务/OM支持多个相同业务关联
